core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -1563,13 +1564,61 @@ trait EntityComponent {
       Entity(entityRecord.name, entityRecord.entityType, attributes)
 
     def unmarshalEntities(
-      entityAttributeRecords: Seq[EntityAndAttributesResult]
+      entityAttributeRecords: Seq[EntityAndAttributesResult],
+      validateOrdering: Boolean = false
     ): Seq[Entity] =
-      unmarshalEntitiesWithIds(entityAttributeRecords).map { case (_, entity) => entity }
+      unmarshalEntitiesWithIds(entityAttributeRecords, validateOrdering).map { case (_, entity) => entity }
+
+    // tail-recursive method that validates if all entity ids are contiguous within the Seq[EntityAndAttributesResult]
+    // an order of 3,3,3,2,2,2,1,1 is ok
+    // an order of 1,2,3,1 is bad, because the 1 is repeated non-contiguously
```

Comment:
can you clarify the return type in the comment too?  I can't quickly tell if a non-empty return value means stuff is good or bad here :)

Reply:
Good feedback, added comments!